homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -80,6 +81,13 @@ async def _async_show_user_form(
                             self.context.get(CONF_URL, ""),
                         ),
                     ): str,
+                    vol.Optional(
```

Comment:
I suggest adding this option into a separate step as it could confuse the users if they insert an HTTP URL.

Maybe even better would be that we try the first time with validation activated, and if it fails, show a step where the user can turn it off.

Reply:
I'm a somewhat torn on this one. I do think the suggestions make sense, but:

We're essentially doing the second already here: if a verified HTTPS connection fails due to a verification error, we say so and give a hint to retry with verification disabled. I'm not sure if a second separate step would be any more helpful.
Verification is off by default because I'm not aware of a router that would/could have a generally out of the box trusted cert installed, so verification must have been explicitly toggled on by the user, and as such, error should not come as a surprise.

With regards to the confusion with a plain HTTP URL, I'm not sure if that's a big issue. We could amend the help text with something like " (if https connection specified)" to alleviate, but that would mean giving up on using the common string for `verify_ssl`, and thus being possibly a bit inconsistent with other things in the future. (I see `vol.Optional` has a `description`, but it doesn't seem to appear in the UI unless I've missed something.)